assignments/array-methods.js

```diff
@@ -56,28 +56,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+
+
+runners.forEach(getFull => {
+    fullName.push(`${getFull.first_name}, ${getFull.last_name}`)
+})
+
+console.log({fullName});
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map(fName => {
+ return fName.first_name.toUpperCase();
+});
+
 console.log(allCaps); 
 
+
+
+// }
+// runners.map(toCaps);
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(shirts => shirts.shirt_size === "L");
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce(function(accumulator, charity) {
+    return accumulator + charity.donation;
+}, 0)
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Seperate the first and last runners into two seperate arrrays by id number
+let half1 = runners.filter((arrayItem)=>{
+    if(arrayItem.id >= 25){
+        return true;
+    } else {
+        return false;
+    };
+ });
+ console.log(half1);
+ 
+ let half2 = runners.filter((arrayItem)=>{
+    if(arrayItem.id <= 25){
+        return true;
+    } else {
+        return false;
+    };
+ });
+ console.log(half2);
+
+ //Problem 2
+
+function mailBlast(runners){
+    console.log(`${runners.id}, ${runners.email}`)
+}
+runners.forEach(mailBlast);
 
-// Problem 2
+// Problem 3
+//Donors over $200 for a special thank you
+let hiRoller = runners.filter((arrayItem) =>{
+    if(arrayItem.donation >= 200){
+        return true;
+    } else {
+        return false;
+    };
+})
+console.log(hiRoller);
```

Comment:
So if your are using a filter you don't need an if else statement. You can filter through the array and just return every object with more than 200 in donations. like so: <h3>let hiRoller = runners.filter((arrayItem) =>  arrayItem.donation >= 200);</h3>